piptools/repositories/pypi.py

```diff
@@ -452,7 +457,7 @@ def _setup_logging(self) -> None:
         logger = logging.getLogger()
         for handler in logger.handlers:
             if handler.name == "console":  # pragma: no branch
-                assert isinstance(handler, logging.StreamHandler)
+                assert isinstance(handler, PipStreamHandler)
```

Comment:
```suggestion
                if not isinstance(handler, PipStreamHandler):
                    raise Exception('A compatible log handler could not be found')
```
@wbolster Thoughts? Not sure what this should say.

Reply:
i guess it'll do, but it's not actionable unless you know about `pip-tools` and `pip` internals, and this may show up to many more people, no?

perhaps something more actionable like ‘No compatible log handler found; pip internals have changed’  (not sure i like it, tbh)

and perhaps `RuntimeError`?

Reply:
have you seen this though? https://github.com/jazzband/pip-tools/pull/1565/files#r797632859

Reply:
2250321 should resolve this

Reply:
nice 👍🏼

Reply:
@di thanks, and sorry for crossing the streams. I was looking at it this morning before I noticed this PR